### ISSUE TITLE
twisted.python.versions support for SVN 1.5 and 1.6

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#3511]
|<img alt="feisley's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| feisley reported|
|-|-|
|Trac ID|trac#3511|
|Type|defect|
|Created|2008-10-29 07:05:12Z|

The SVN version parser for twisted.python.versions needs to be updated to support the new SVN version formats:

SVN 1.5 -> format 9

SVN 1.6 -> format 10

Both of these versions follow the specifications as format 8 that is currently implemented. 

A simple fix would simply to make new methods to handle the new versions based on the existing compatible version:

```
_parseSVNEntries_9 = _parseSVNEntries_8
_parseSVNEntries_10 = _parseSVNEntries_8
```

However, this seems suboptimal and there is most likely a cleaner way to go about this.

Attachments:

* [versions-format.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/ba4/ba42347b3fd300dc9636788b2da5ce32ccb010d2/b29f010ee41f39c0d98420628f2def810c9f3535.patch) (2527 bytes) - added by feisley on 2008-10-29 14:10:30Z - Patch for versions support
* [versions-format.2.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/ba4/ba42347b3fd300dc9636788b2da5ce32ccb010d2/187a4c00620270df9276ee753ffb66c6d321c2e5.patch) (2802 bytes) - added by exarkun on 2008-10-31 14:28:55Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__3511 3511
type__defect defect
reporter__feisley feisley
priority__normal normal
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__easy easy
time__1225263912000000 1225263912000000
changetime__1226769648000000 1226769648000000
version__None None
owner__ 
cc__exarkun
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun commented|
|-|-|

Seems like a fine approach to me.  Why is it suboptimal?

**Comment by twisted-trac** [comment]
|<img alt="feisley's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_2"></a>|feisley commented|
|-|-|

Perhaps not suboptimal... Maybe just not as elegant as I was hoping for (though things usually are)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_4"></a>|@exarkun set owner to feisley|
|-|-|

The patch has a couple minor mistakes, probably due to copy/pasting.  I'll attach a modified version which fixes those.

A bigger issue is that I haven't been able to find docs for format 9 or format 10 anywhere.  Can you provide a pointer for that?

**Comment by twisted-trac** [comment]
|<img alt="feisley's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_5"></a>|feisley commented|
|-|-|

I as well was unable to actually find any docs that specifically described the format of those two versions. Due to the limited nature of the query being performed here, I simply inspected the "entries" file to see that the lines did in fact match.

**Comment by twisted-trac** [comment]
|<img alt="feisley's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|feisley set owner to @exarkun|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_7"></a>|@exarkun commented|
|-|-|

Okay.  Since svn 1.6 isn't actually released yet, how confident are we that the format the development branch is using now will be the same as the format it's released with?

**Comment by twisted-trac** [comment]
|<img alt="feisley's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_8"></a>|feisley commented|
|-|-|

While they could still change the format used in 1.6, historically they have never changed a format spec once it has been numbered. For example, in the event they would want a new format for 1.6 they would likely make a new format 11.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_9"></a>|@exarkun set status to `closed`|
|-|-|

(In [25349]) Apply versions-format.2.patch

Author: feisley
Reviewer: exarkun
Fixes: [#3511](https://github.com/twisted/twisted/issues/3511)

Add support to `twisted.python.versions` for parsing SVN format 9 and 10 entries
files.  These are the versions used by subversion 1.5 and (speculatively) 1.6
respectively.  They do not differ from format 8 as far as `Version` is concerned,
so the version 8 parser is re-used.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
